### PR TITLE
Task/refactor mine assocated resources

### PR DESCRIPTION
**Pull request by Jsyro**
data driven mine type codes, mine type post/delete handles mine_type_details. 

Still working on some backend tests.

### CODE REVIEW

File: python-backend/app/api/mines/mine/resources/mine_type.py [hunk 1]
```diff
@@ -7,53 +8,65 @@
...
+        
+        current_app.logger.debug(mine_type.mine_tenure_type)
+        for d_code in mine_disturbance_code:
+            MineTypeDetail.create(mine_type, mine_disturbance_code=d_code)
 
+        for c_code in mine_commodity_code:
+            MineTypeDetail.create(mine_type, mine_commodity_code=c_code)
+
+        mine_type.save()
+        current_app.logger.debug(mine_type.mine_tenure_type_code)
```
Comment (nayr974) [inline python-backend/app/api/mines/mine/resources/mine_type.py]: Is this test code or intended to go to prod?

File: python-backend/app/api/mines/mine/models/mine_type_detail.py [hunk 2]
```diff
@@ -2,54 +2,73 @@
...
+                                      db.ForeignKey('mine_disturbance_code.mine_disturbance_code'))
+    mine_commodity_code = db.Column(db.String(2),
+                                    db.ForeignKey('mine_commodity_code.mine_commodity_code'))
+
     active_ind = db.Column(db.Boolean, nullable=False, default=True)
 
     def __repr__(self):
         return '<MineTypeDetail %r>' % self.mine_type_detail_xref_guid
 
-    def json(self):
```
Comment (nayr974) [inline python-backend/app/api/mines/mine/models/mine_type_detail.py]: DIE DIE DIE DIE DIE 💯 🥇

File: python-backend/app/api/mines/mine/models/mine_type.py [hunk 3]
```diff
@@ -47,7 +41,8 @@ def find_by_guid(cls, _id):
         uuid.UUID(_id, version=4)
         return cls.query.filter_by(mine_type_guid=_id).first()
 
-    @classmethod
-    def expire_record(cls, record):
-        record.active_ind = False
-        record.save()
+    def expire_record(self):
```
Comment (nayr974) [inline python-backend/app/api/mines/mine/models/mine_type.py]: "expire record" seems awkward, why not call it "inactivate" which describes the logic - setting active = false.

File: python-backend/app/api/mines/mine/models/mine_type.py [hunk 4]
```diff
@@ -1,43 +1,37 @@
...
 from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.schema import FetchedValue
 from ....utils.models_mixins import AuditMixin, Base
 from app.extensions import db
 
 
 class MineType(AuditMixin, Base):
     __tablename__ = "mine_type"
-    mine_type_guid = db.Column(UUID(as_uuid=True), primary_key=True)
+    mine_type_guid = db.Column(UUID(as_uuid=True), primary_key=True, server_default=FetchedValue())
```
Comment (nayr974) [inline python-backend/app/api/mines/mine/models/mine_type.py]: 👍

File: python-backend/app/api/mines/mine_api_models.py [hunk 5]
```diff
@@ -37,6 +37,22 @@ def format(self, value):
     'description': fields.String,
 })
 
+MINE_COMMODITY_CODE_MODEL = api.model(
+    'MineCommodityCodeModel', {
+        'mine_commodity_code': fields.String,
+        'description': fields.String,
+        'active_ind': fields.Boolean,
```
[the remark this answers is not in the document]
Comment (Jsyro) [reply python-backend/app/api/mines/mine_api_models.py]: Yup.

**Pull request merged**